src/Conversion/ONNXToKrnl/Quantization/QuantizeLinear.cpp

```diff
@@ -71,13 +74,24 @@ void emitQuantizationLinearScalarParameters(ConversionPatternRewriter &rewriter,
   DimsExpr outputAF;
   outputAF.emplace_back(zero);
 
+  Value oneOverScale;
+  bool useOneOverScale =
+      !DISABLE_FAST_MATH_FOR_QL && isa<FloatType>(inputElementType);
+  if (useOneOverScale) {
+    Value one = create.math.constant(inputElementType, 1.0);
+    oneOverScale = create.math.div(one, scale);
```

Comment:
I wonder whether we can have a better performance if we prepare a vector here instead of a scalar. I guess in the loop `create.math.mul(x, oneOverScale);` will splat `oneOverScale` whose type is of scalar.

Reply:
No Splat in the innermost loop, they are migrated out.

```mlir
    scf.for %arg3 = %c0_6 to %c65536_7 step %c16 {
      %8 = vector.load %reshape[%arg3] : memref<65536xf32>, vector<16xf32>
      %9 = arith.mulf %8, %6 : vector<16xf32>
      %10 = vector.shape_cast %9 : vector<16xf32> to vector<4x4xf32>
      %11 = vector.extract %10[0] : vector<4xf32> from vector<4x4xf32>
      %12 = "krnl.round_even"(%11) : (vector<4xf32>) -> vector<4xf32>
      %13 = vector.insert %12, %10 [0] : vector<4xf32> into vector<4x4xf32>
      %14 = vector.extract %10[1] : vector<4xf32> from vector<4x4xf32>
      %15 = "krnl.round_even"(%14) : (vector<4xf32>) -> vector<4xf32>
      %16 = vector.insert %15, %13 [1] : vector<4xf32> into vector<4x4xf32>
      %17 = vector.extract %10[2] : vector<4xf32> from vector<4x4xf32>
      %18 = "krnl.round_even"(%17) : (vector<4xf32>) -> vector<4xf32>
      %19 = vector.insert %18, %16 [2] : vector<4xf32> into vector<4x4xf32>
      %20 = vector.extract %10[3] : vector<4xf32> from vector<4x4xf32>
      %21 = "krnl.round_even"(%20) : (vector<4xf32>) -> vector<4xf32>
      %22 = vector.insert %21, %19 [3] : vector<4xf32> into vector<4x4xf32>
      %23 = vector.shape_cast %22 : vector<4x4xf32> to vector<16xf32>
      %24 = arith.addf %23, %7 : vector<16xf32>
      %25 = arith.maxnumf %24, %cst_0 : vector<16xf32>
      %26 = arith.minnumf %25, %cst : vector<16xf32>
      %27 = arith.fptoui %26 : vector<16xf32> to vector<16xi32>
      %28 = arith.trunci %27 : vector<16xi32> to vector<16xi8>
      %29 = builtin.unrealized_conversion_cast %28 : vector<16xi8> to vector<16xui8>
      vector.store %29, %reshape_5[%arg3] : memref<65536xui8>, vector<16xui8>
    }
```

Reply:
Great! Thanks!